oi/CodeGen.cpp

```diff
@@ -162,6 +163,82 @@ void genDecls(const TypeGraph& typeGraph, std::string& code) {
   }
 }
 
+/*
+ * Generates a declaration for a given fully-qualified type.
+ *
+ * e.g. Given "nsA::nsB::Foo"
+ *
+ * The folowing is generated:
+ *   namespace nsA {
+ *   namespace nsB {
+ *   struct Foo;
+ *   } // nsB
+ *   } // nsA
+ */
+void declareThriftStruct(const std::string& name, std::string& code) {
+  if (auto pos = name.find("::"); pos != name.npos) {
```

Comment:
Could we change this to use nested namespaces e.g. `namespace nsA::nsB { struct Foo; }`? Should be quite easy with an `.rfind("::")` and also remove the recursion.